site/content/dev-guide/how-tos/implement-an-adapter.md

```diff
@@ -0,0 +1,112 @@
+
+# Prerequisites and Assumptions
+* It is assumed that the developer(s) have familiarized themselves with the  [NoSqlBench core concepts](https://docs.nosqlbench.io/introduction/core-concepts/) provided in the nosqlbench documentation.
+* The ability to code in Java is required. NoSqlBench makes extensive use of lambda functionality. It is a requirement to understand and be comfortable using these principles to develop an adapter.
+* This guide is not specific to any particular client. It is assumed that the developer(s) have familiarity with the client they are developing an adapter for.
+
+# Internal Dependencies
+The nb project as a whole is composed of numerous modules and packages. For the purpose of customer adapter implementation there are only two required; adapters-api and nb-annotations.
+
+	<dependency>
+		<groupId>io.nosqlbench</groupId>
+     	<artifactId>adapters-api</artifactId>
+     	<version>${revision}</version>
+     	<scope>compile</scope>
+	</dependency>
+		
+	<dependency>
+        <groupId>io.nosqlbench</groupId>
+        <artifactId>nb-annotations</artifactId>
+        <version>${revision}</version>
+        <scope>compile</scope>
+	</dependency>
+	
+The adapters-api and nb-annotations  modules contain all of the base classes, interfaces and annotations required to implement a new adapter and flesh out its functionality.
+
+* Adapters - Every new adapter will, of course, include an Adapter class. This class will be a descendent of the BaseDriverAdapter class found here. Nosqlbench uses the Java Service Provider Interface(SPI) to define service providers. If you don’t know what that is, don't worry about it now. The tl;dr is that extending the BaseDriverAdapter and annotating your driver adapter class with the Service annotation will result in that class being added to META-INF/services under the specified class name, making it available as a service provider. Nosqlbench takes care of all the rest!
+* Ops - Nosqlbench is built around the ability to define operations (“Ops”) in a generic fashion (the op template) and translate them into actions. All of the basic plumbing for this is handled in the adapters-api module, from the yaml file loader to the definition of the various types of ops available to implement. Along with the Op types the adapters-api package defines the base classes and interfaces for Op Dispensers, which do the work of creating the Op, and the Op Mappers, which map the appropriate functionality based on the Op type presented. We’re going to get into the details of the available implementations in the next section.
+
+# Your First Adapter
```

Comment:
Formatting and update of numbering needed in the section below.

Reply:
corrected